browser/fragments/ably-epilogue.js

```diff
@@ -1,16 +1,35 @@
-if(typeof Realtime !== 'undefined') {
-	Ably.msgpack = msgpack;
-	Ably.Rest = Rest;
-	Ably.Realtime = Realtime;
-	Realtime.ConnectionManager = ConnectionManager;
-	Realtime.BufferUtils = Rest.BufferUtils = BufferUtils;
-	if(typeof(Crypto) !== 'undefined') Realtime.Crypto = Rest.Crypto = Crypto;
-	Realtime.Defaults = Rest.Defaults = Defaults;
-	Realtime.Http = Rest.Http = Http;
-	Realtime.Utils = Rest.Utils = Utils;
-	Realtime.Http = Rest.Http = Http;
-	Realtime.Message = Rest.Message = Message;
-	Realtime.PresenceMessage = Rest.PresenceMessage = PresenceMessage;
-	Realtime.ProtocolMessage = Rest.ProtocolMessage = ProtocolMessage;
-}
+	if(typeof Realtime !== 'undefined') {
+		Ably.msgpack = msgpack;
+		Ably.Rest = Rest;
+		Ably.Realtime = Realtime;
+		Realtime.ConnectionManager = ConnectionManager;
+		Realtime.BufferUtils = Rest.BufferUtils = BufferUtils;
+		if(typeof(Crypto) !== 'undefined') Realtime.Crypto = Rest.Crypto = Crypto;
+		Realtime.Defaults = Rest.Defaults = Defaults;
+		Realtime.Http = Rest.Http = Http;
+		Realtime.Utils = Rest.Utils = Utils;
+		Realtime.Http = Rest.Http = Http;
+		Realtime.Message = Rest.Message = Message;
+		Realtime.PresenceMessage = Rest.PresenceMessage = PresenceMessage;
+		Realtime.ProtocolMessage = Rest.ProtocolMessage = ProtocolMessage;
+	}
 }).call({});
+
+/* CommonJS support */
+if (typeof exports === "object" && exports) {
+	for (var obj in window.Ably) {
+		if (window.Ably.hasOwnProperty(obj)) {
+			exports[obj] = window.Ably[obj];
+		}
+	}
+	/* SystemJS support for default exports to be added to the root of the module
+	   https://github.com/frankwallis/plugin-typescript/issues/185 */
+	exports.__esModule = true;
+}
+
+/* AMD support */
+if (typeof define === "function" && define.amd) {
+  define("ably", [], function() {
+    return Ably;
+  });
+}
```

Comment:
> Well I know our approach before was to create separate versions, but is that necessary? Why not add CommonJS support when CommonJS is being used?

If that works, sure. I'm a bit fuzzy on how commonjs/amd/etc work, so happy to leave to your judgement.

But if we're doing that, we should probs remove the separate ably-commonjs.js (or rather, remove the separate generator and make it another version of the same file) -- we don't want to have  two different versions of ably-js which both support all module systems but do so in slightly different ways...

(don't mind if you'd rather merge this now and make an issue to sort it out later if you prefer)